libs/@guardian/identity-auth-frontend/src/index.ts

```diff
@@ -0,0 +1,66 @@
+import { IdentityAuth } from '@guardian/identity-auth';
+import type { CustomClaims } from '@guardian/identity-auth';
+
+type FrontendIdTokenClaims = CustomClaims & {
+	email: string;
+	braze_uuid: string;
+};
+
+export type FrontendIdentityAuth = IdentityAuth<never, FrontendIdTokenClaims>;
+
+const getStage = (isDev: boolean, stage: string) => (isDev ? 'DEV' : stage);
+
+const getIssuer = (stage: string) =>
+	stage === 'PROD'
+		? 'https://profile.theguardian.com/oauth2/aus3xgj525jYQRowl417'
+		: 'https://profile.code.dev-theguardian.com/oauth2/aus3v9gla95Toj0EE0x7';
+
+const getClientId = (stage: string) =>
+	stage === 'PROD' ? '0oa79m1fmgzrtaHc1417' : '0oa53x6k5wGYXOGzm0x7';
+
+const getRedirectUri = (stage: string) => {
+	switch (stage) {
+		case 'PROD':
+			return 'https://www.theguardian.com/';
+		case 'CODE':
+			return 'https://m.code.dev-theguardian.com/';
+		case 'DEV':
+		default:
+			return 'http://localhost:3030/';
+	}
+};
+
+export const getIdentityAuth = () => {
+	const { guardian } = window;
+	const { config } = guardian ?? {};
+	const { isDev, stage } = guardian?.config ?? {};
+
+	if (!guardian || !config || !isDev || !stage) {
+		throw new Error('window.guardian has not yet been initialized');
+	}
+
+	const stageOrDev = getStage(isDev, stage);
+
+	if (!guardian.identityAuth) {
```

Comment:
Instead of saving the instance of IdentityAuth to the window I could have stored it inside the module as a global field, however I was worried that differences between module loading of Frontend vs Commercial might cause the module to be imported/created multiple times and have different instances of IdentityAuth

Reply:
We use [conditional assignment with `||=`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR_assignment) in [`@guardian/consent-management-platform`](https://github.com/guardian/consent-management-platform/blob/e768b95da95ded9c6320247cf5ddebb9beeb19a4/src/index.ts#L91-L103)